### ISSUE TITLE
Rolling csv client daily

### ISSUE THREAD

**Issue by MichaelWS** [Aurora-Team/BitcoinExchangeFH#71]
I am recording data for future use and wanted to see if you were open to a pull request that would roll and compress the csv data when utc date moves to the next day

**Comment by gavincyi** [comment]
I think it is a good idea but I am doubtful whether it is too difficult to achieve a generic functionality. It may involve:

- Access right to folders or other faults which is due to fail to overwrite the file
- Compression takes time. It may lead to the rollover starts at different time among the instruments.

I'll think about its approach to suit most users.

Anyway thanks a lot for your suggestion.

**Comment by MichaelWS** [comment]
Gavin, I am happy to issue a pull request when you have an idea how to generalize.  I suppose you would need a cross platform compression program.   As for start time, I would just use subprocess to call so it would be no issue on start time
https://github.com/MichaelWS/BitcoinExchangeFH/blob/master/befh/file_client.py#L66

**Comment by gavincyi** [comment]
Does [gzip](https://docs.python.org/3/library/gzip.html) work for Windows environment? I haven't used it before but it seems a good fit.

**Comment by MichaelWS** [comment]
well it does but its much slower than just calling gzip on a file.

much faster/easier  to do something like this to add windows support.
https://github.com/MichaelWS/BitcoinExchangeFH/blob/master/befh/file_client.py#L69

**Comment by gavincyi** [comment]
Sorry that I was busy in these days and did not get you back on it.

I am happy if you issue a pull request and I will review soon. Appreciate a lot for your effort!

Honestly I prefer gzip with multiprocessing/other non-blocking modules than subprocess. I think why python is prevailing is that developers/users do not need to handle the coding/behavior in different platforms (OS). It's the magic of Python. :)

**Comment by MichaelWS** [comment]
Yes, I guess I am from more of a performance background and the gzip is
slow.  I will give it a go this weekend

On Aug 15, 2017 21:48, "Gavin Chan" <notifications@github.com> wrote:

Sorry that I was busy in these days and did not get you back on it.

I am happy if you issue a pull request and I will review soon. Appreciate a
lot for your effort!

Honestly I prefer gzip with multiprocessing/other non-blocking modules than
subprocess. I think why python is prevailing is that developers/users do
not need to handle the coding/behavior in different platforms (OS). It's
the magic of Python. :)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Aurora-Team/BitcoinExchangeFH/issues/71#issuecomment-322639840>,
or mute the thread
